rotkehlchen/db/upgrades/v48_v49.py

```diff
@@ -48,4 +51,37 @@ def _fix_zksynclite_swaps_schema(write_cursor: 'DBCursor') -> None:
             insert_columns='tx_id, from_asset, from_amount, to_asset, COALESCE(to_amount, "0")',
         )
 
+    @progress_step(description='Migrating solana assets to CAIPS format.')
+    def _migrate_solana_assets_to_caips(write_cursor: 'DBCursor') -> None:
+        dir_path = Path(__file__).resolve().parent.parent.parent
+        if (csv_file := dir_path / 'data' / 'solana_tokens_data.csv').exists() is False:
+            log.debug(f'Missing CSV file at {csv_file}. Skipping solana assets migration.')
+            return
+
+        asset_updates = []
```

Comment:
this is also similar to the one in the globaldb right? Wouldn't it be better to abstract it??

Reply:
i would have done that but we don't abstract functions in db upgrade if my memory serves me well.